torch_em/self_training/mean_teacher.py

```diff
@@ -196,11 +197,13 @@ def _train_epoch_unsupervised(self, progress, forward_context, backprop):
 
             teacher_input, model_input = xu1, xu2
 
+            with torch.no_grad():
```

Comment:
You should also add the `forward_context()` to the `with` statement. (See also above).